src/mqt/core/qiskit_utils/imports.py

```diff
@@ -0,0 +1,337 @@
+                f = float(factor_no_operator)
+                coeff *= 1.0 / f if is_div else f
+            else:
+                var = factor_no_operator
+
+        if var == "":
+            expr += coeff
+        else:
+            is_const = False
+            expr += Term(sign * coeff, Variable(var))
+
+    if is_const:
+        return expr.constant
+    return expr
+
+
+def _add_operation(
+    mqt_computation: QuantumComputation,
+    type_: OpType,
+    qargs: tuple[Qubit] | list[Qubit],
+    params: list[float | ParameterExpression],
+    qubit_map: dict[Qubit, int],
+) -> None:
+    qubits = [qubit_map[qubit] for qubit in qargs]
+    target = qubits.pop()
+    parameters = [_parse_symbolic_expression(param) for param in params]
+    controls = {Control(qubit) for qubit in qubits}
+    if all(isinstance(parameter, float) for parameter in parameters):
+        float_parameters = [cast(float, parameter) for parameter in parameters]
+        mqt_computation.append_operation(
+            StandardOperation(mqt_computation.n_qubits, controls, target, type_, float_parameters)
+        )
+    else:
+        mqt_computation.append_operation(
+            SymbolicOperation(mqt_computation.n_qubits, controls, target, type_, parameters)
+        )
+        for parameter in parameters:
+            mqt_computation.add_variable(parameter)
+
+
+def _add_two_target_operation(
+    mqt_computation: QuantumComputation,
+    type_: OpType,
+    qargs: tuple[Qubit] | list[Qubit],
+    params: list[float | ParameterExpression],
+    qubit_map: dict[Qubit, int],
+) -> None:
+    qubits = [qubit_map[qubit] for qubit in qargs]
+    target1 = qubits.pop()
+    target2 = qubits.pop()
+    parameters = [_parse_symbolic_expression(param) for param in params]
+    controls = {Control(qubit) for qubit in qubits}
+    if all(isinstance(parameter, float) for parameter in parameters):
+        float_parameters = [cast(float, parameter) for parameter in parameters]
+        mqt_computation.append_operation(
+            StandardOperation(mqt_computation.n_qubits, controls, target1, target2, type_, float_parameters)
+        )
+    else:
+        mqt_computation.append_operation(
+            SymbolicOperation(mqt_computation.n_qubits, controls, target1, target2, type_, parameters)
+        )
+        for parameter in parameters:
+            mqt_computation.add_variable(parameter)
+
+
+def _get_logical_qubit_indices(mqt_computation: QuantumComputation, layout: Layout) -> dict[Qubit, int]:
+    registers = layout.get_registers()
+    logical_qubit_index = 0
+    logical_qubit_indices = {}
+    ancilla_register = None
+
+    for register in registers:
+        if register.name == "ancilla":
+            ancilla_register = register
+            continue
+
+        size = register.size
+        for physical_qubit_index in range(size):
+            logical_qubit_indices[Qubit(register, physical_qubit_index)] = logical_qubit_index
+            logical_qubit_index += 1
```

Comment:
As above, why not just iterate over the register itself?

Reply:
I think i just need the index here